src/main/java/software/amazon/awssdk/crt/io/HostResolver.java

```diff
@@ -36,6 +36,57 @@ protected void releaseNativeHandle() {
         }
     }
 
+    /*
+     * Static interface for access to a default, lazily-created host resolver for users who don't
+     * want to deal with the associated resource management.  Client bootstraps will use this host resolver
+     * if they are passed a null value.
+     */
+
+    /**
+     * Sets the max number of cached host entries for the static default resolver, if it's ever created/used. Has no
+     * effect if the static default host resolver has already been created.
+     *
+     * @param maxEntries maximum number of host entries cached
+     */
+    public static void setStaticDefaultMaxEntries(int maxEntries) {
```

Comment:
same. `setStaticDefault(HostResolver)` seems simpler

Reply:
Nothing stops us from adding those later, but this is the "I don't want to manage crt resources" interface and presumably we would be taking refs to passed in objects which implies the user still kind of needs to think about it too.